test/components/generators/test_hugging_face_local_generator.py

```diff
@@ -4,10 +4,16 @@
 import pytest
 import torch
 from transformers import PreTrainedTokenizerFast
-from haystack.utils.auth import Secret
 
 from haystack.components.generators.hugging_face_local import HuggingFaceLocalGenerator, StopWordsCriteria
 from haystack.utils import ComponentDevice
+from haystack.utils.auth import Secret
+from haystack.utils.hf import HFTokenStreamingHandler
+
+
+# used to test serialization of streaming_callback
+def streaming_callback_handler(x):
```

Comment:
I think it makes more sense to define this inside the `test_run_with_streaming(self)` itself, since it's only used there

Reply:
Good suggestion @davidsbatista - I would but the serialization mechanism doesn't see the namespace of the function IIRC.

Reply:
Nevermind, it can be done, thanks for the suggestion